cypapi/cypapi.pyx

```diff
@@ -328,17 +328,24 @@ cdef class PyPAPI_EventSet:
             values[i] = vals[i]
         PyMem_Free(vals)
 
-    def list_events(self):
+    def list_events(self, probe = False):
         cdef int num_events = self.num_events()
-        cdef int *evts = <int *> PyMem_Malloc(num_events * sizeof(int))
-        if not evts:
-            raise Exception('Failed to allocate array')
-        cdef int number = 0
-        cdef int papi_errno = PAPI_list_events(self.event_set, evts, &number)
-        if papi_errno != PAPI_OK:
-            raise Exception(f'PAPI Error {papi_errno}: PAPI_list_events failed.')
-        result = [evts[i] for i in range(num_events)]
-        PyMem_Free(evts)
+        cdef int *evts, papi_errno
```

Comment:
If based on Friday's discussion that this implementation of `list_events(...)` is what we move forward with update code such that the declaration of `num_events` is moved down to this line.